internal/verifier/sigstore/container/container.go

```diff
@@ -26,37 +26,96 @@ import (
 	"encoding/pem"
 	"errors"
 	"fmt"
+	"net/http"
+	"strings"
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/crane"
 	"github.com/google/go-containerregistry/pkg/name"
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	containerdigest "github.com/opencontainers/go-digest"
+	"github.com/rs/zerolog"
 	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
 	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 	protorekor "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
 	"github.com/sigstore/sigstore-go/pkg/bundle"
+	"github.com/sigstore/sigstore-go/pkg/fulcio/certificate"
 	"github.com/sigstore/sigstore-go/pkg/verify"
+	"google.golang.org/protobuf/encoding/protojson"
 
 	"github.com/stacklok/minder/internal/util"
 	pb "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+	provifv1 "github.com/stacklok/minder/pkg/providers/v1"
 )
 
+var (
+	// ErrOciImageSignatureNotFound is returned when the OCI image signature is not found
+	ErrOciImageSignatureNotFound = errors.New("OCI image signature not found")
+	// ErrAuthNotProvided is returned when the selected authentication is not provided
+	ErrAuthNotProvided = errors.New("selected auth method not provided")
+)
+
+// AuthMethod is an option for containerAuth
+type AuthMethod func(auth *containerAuth)
+
+// containerAuth is the authentication for the container
+type containerAuth struct {
+	accessToken string
+	ghClient    provifv1.GitHub
+}
+
+func newContainerAuth(authOpts ...AuthMethod) *containerAuth {
+	var auth containerAuth
+	for _, opt := range authOpts {
+		opt(&auth)
+	}
+	return &auth
+}
+
+// WithAccessToken sets the access token as an authentication option we want to use during verification
+func WithAccessToken(accessToken string) AuthMethod {
+	return func(auth *containerAuth) {
+		auth.accessToken = accessToken
+	}
+}
+
+// WithGitHubClient sets the GitHub client as an authentication option we want to use during verification
+func WithGitHubClient(ghClient provifv1.GitHub) AuthMethod {
+	return func(auth *containerAuth) {
+		auth.ghClient = ghClient
+	}
+}
+
 // Verify verifies a container artifact using sigstore
-func Verify(_ context.Context, sev *verify.SignedEntityVerifier, accessToken, registry, owner, artifact, version string) (
-	[]byte, []byte, error) {
+func Verify(
+	ctx context.Context,
+	sev *verify.SignedEntityVerifier,
+	registry, owner, artifact, version string,
+	authOpts ...AuthMethod,
+) ([]byte, []byte, error) {
+
 	// create a default verification result
 	params := newVerifyResult(BuildImageRef(registry, owner, artifact, version))
 
+	auth := newContainerAuth(authOpts...)
+
 	// construct the bundle
-	err := bundleFromOCIImage(params, newGithubAuthenticator(owner, accessToken))
+	// TODO: handle no access token?
```

Comment:
I think we should error out, we should always have a token

Reply:
done, tested by running `mindev container verify` with no token.